lib/python/qmk/commands.py

```diff
@@ -0,0 +1,57 @@
+"""Functions that build make commands
+"""
+import json
+import qmk.keymap
+
+def create_make_command(keyboard, keymap, target=None):
+    """Create a make compile command
+
+    Args:
+        keyboard
+            The path of the keyboard, for example 'plank'
+
+        keymap
+            The name of the keymap, for example 'algernon'
+
+        target
+            Usually a bootloader.
+
+    Returns:
+        A command that can be run to make the specified keyboard and keymap
+    """
+    if target is None:
+        return ['make', ':'.join((keyboard, keymap))]
+    return ['make', ':'.join((keyboard, keymap, target))]
+
+def parse_configurator_json(configurator_filename):
+    """Open and parse a configurator json export
+    """
+    file = open(configurator_filename)
```

Comment:
Should we do some type of validation here so that we know this is an actual configurator file (there's probably keys and such we can check for the presence of), otherwise we're going to error out hard later down the code when we start referencing keys that may not exist if we get a bad/unexpected file

Reply:
skullydazed stated on discord "that's out of scope for this PR, in the future we'll have defined schemas for the file". So Im going to leave this how it is for now. Thank you for the feedback though!